### PR TITLE
Improve debug logging in cloud foundry container tagger

### PR DESCRIPTION
**Pull request by sarah-witt**
### What does this PR do?

Improves the logging in the cloud foundry container tagger

### Motivation

In some cases, it is helpful for debugging to know what event has causes a container tagger retry event. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/cloudfoundry/containertagger/container_tagger.go [hunk 1]
```diff
@@ -121,8 +123,9 @@ func (c *ContainerTagger) processEvent(ctx context.Context, evt workloadmeta.Eve
 		go func() {
 			var exitCode int
 			var err error
-			for retry := 0; retry < c.retryCount; retry++ {
-				log.Infof("Updating tags in container `%s` retry #%d", containerID, retry)
+			for attempt := 1; attempt < c.retryCount; attempt++ {
```
Comment (NouemanKHAL) [inline pkg/cloudfoundry/containertagger/container_tagger.go:126]:
```suggestion
			for attempt := 1; attempt <= c.retryCount; attempt++ {
```

**Review by NouemanKHAL — CHANGES_REQUESTED** [review]
Few nits, also I guess it'd be better if we change the log level of line 133:
```
log.Debugf("Successfully updated tags in container `%s`", containerID)
```
to:
```
log.Infof("Successfully updated tags in container `%s`", containerID)
````

File: pkg/cloudfoundry/containertagger/container_tagger.go (hunk 1 above)
Comment (sarah-witt) [reply pkg/cloudfoundry/containertagger/container_tagger.go:126]: Ha good catch!

File: pkg/cloudfoundry/containertagger/container_tagger.go [hunk 2]
```diff
@@ -121,8 +123,9 @@ func (c *ContainerTagger) processEvent(ctx context.Context, evt workloadmeta.Eve
...
+				log.Infof("Updating tags in container `%s` attempt #%d", containerID, attempt)
+				log.Debugf("Update attempt #%d for event %d", attempt, eventTimestamp)
```
[the remark this answers is not in the document]
Comment (sarah-witt) [reply pkg/cloudfoundry/containertagger/container_tagger.go:128]: I preferred it this way so we'd always get an info line about updating tags. But I agree it does pollute the logs more.Is that why you'd like to compress it to one line?

File: pkg/cloudfoundry/containertagger/container_tagger.go [hunk 3]
```diff
@@ -87,10 +87,12 @@ func (c *ContainerTagger) Start(ctx context.Context) {
...
 func (c *ContainerTagger) processEvent(ctx context.Context, evt workloadmeta.Event) error {
-	containerID := evt.Entity.GetID().ID
-
+	entity := evt.Entity
+	containerID := entity.GetID().ID
 	if evt.Type == workloadmeta.EventTypeSet {
-		storeContainer := evt.Entity.(*workloadmeta.Container)
+		eventTimestamp := time.Now().UnixNano()
+		storeContainer := entity.(*workloadmeta.Container)
+		log.Debugf("Processing Event (id %d): %+v", eventTimestamp, storeContainer)
```
Comment (NouemanKHAL) [inline pkg/cloudfoundry/containertagger/container_tagger.go:95]: Same here, let's create a `eventID := fmt.Sprintf("%s-%d", containerId, eventTimestamp)` and use it everywhere.

**Review by NouemanKHAL — APPROVED** [review]

**Pull request merged**